util/encoding/encoding.go

```diff
@@ -297,3 +293,49 @@ func DecodeVarUint64Decreasing(b []byte) ([]byte, uint64) {
 	leftover, v := DecodeVarUint64(b)
 	return leftover, ^v
 }
+
+const (
+	escape = 0x00
+	term   = 0x01
+	null   = 0xff
+)
+
+// EncodeBytes ...
+func EncodeBytes(b []byte, data []byte) []byte {
+	copyStart := 0
+	for i, v := range data {
+		if v == escape {
+			b = append(b, data[copyStart:i]...)
+			b = append(b, escape, null)
+			copyStart = i + 1
+		}
+	}
+	b = append(b, data[copyStart:]...)
+	return append(b, escape, term)
+}
+
+// DecodeBytes ...
+func DecodeBytes(b []byte) ([]byte, []byte) {
+	var r []byte
+	copyStart := 0
+	for i := 0; i < len(b); i++ {
+		v := b[i]
+		if v == escape {
+			v = b[i+1]
```

Comment:
Again this would fail on a simple invalid input of a string with just a single escape character. Add a test for this.